tests/CMakeLists.txt

```diff
@@ -14,6 +14,34 @@
 # limitations under the License.
 # ~~~
 
+set(vulkan_core_header_file "${PROJECT_SOURCE_DIR}/include/vulkan/vulkan_core.h")
+if (NOT EXISTS ${vulkan_core_header_file})
+    message(FATAL_ERROR "Couldn't find vulkan_core.h!")
+endif()
+
+file(READ ${vulkan_core_header_file} ver)
+
+# Get the major/minor version
+if (ver MATCHES "#define[ ]+VK_HEADER_VERSION_COMPLETE[ ]+VK_MAKE_API_VERSION\\([ ]*[0-9]+,[ ]*([0-9]+),[ ]*([0-9]+),[ ]*VK_HEADER_VERSION[ ]*\\)")
+    set(VK_VERSION_MAJOR "${CMAKE_MATCH_1}")
+    set(VK_VERSION_MINOR "${CMAKE_MATCH_2}")
+else()
+    message(FATAL_ERROR "Couldn't get major/minor version")
+endif()
+
+# Get the patch version
+if (ver MATCHES "#define[ ]+VK_HEADER_VERSION[ ]+([0-9]+)")
+    set(VK_HEADER_VERSION "${CMAKE_MATCH_1}")
+else()
+    message(FATAL_ERROR "Couldn't get the patch version")
+endif()
+
+set(VK_VERSION_STRING "${VK_VERSION_MAJOR}.${VK_VERSION_MINOR}.${VK_HEADER_VERSION}")
+
+if (NOT ${VK_VERSION_STRING} STREQUAL ${PROJECT_VERSION})
+    message(FATAL_ERROR "Project version is ${PROJECT_VERSION} but should be ${VK_VERSION_STRING}!")
```

Comment:
Since CI enables testing, this will catch if the version hasn't been properly updated.

Reply:
The error message will look like this:
```
[cmake] CMake Error at tests/CMakeLists.txt:42 (message):
[cmake]   Project version is 1.2.238 but should be 1.3.238!
```